packages/flutter/lib/src/widgets/_web_browser_detection_io.dart

```diff
@@ -0,0 +1,12 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// The widget in this file is an empty mock for non-web platforms. See
+// `_web_browser_detection.dart` for the web implementation.
+
+/// A class that provides information about the current browser when on the web.
+class WebBrowserDetection {
+  /// Whether the current browser is webkit (Safari).
```

Comment:
Should we explain that this throws if called on non-web platforms?

Reply:
Alternatively, could we make this return `false`? We know the dart:io implementation won't be used on Safari, right?

Reply:
Good point, `false` sounds like a reasonable default!